### PR TITLE
test: skip testing EC cofactor generate tests for older FIPS providers

### PR DESCRIPTION
**Pull request by paulidale**
Urgent because extended CIs are failing.  Problem introduced in #25548.

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: test/acvp_test.c [hunk 1]
```diff
@@ -551,10 +557,8 @@ static int dsa_keygen_test(int id)
     size_t priv_len = 0, pub_len = 0;
     const struct dsa_paramgen_st *tst = &dsa_keygen_data[id];
 
-    if (!dsasign_allowed) {
-        TEST_info("DSA keygen test skipped: DSA signing is not allowed");
-        return 1;
-    }
+    if (!dsasign_allowed)
+        return TEST_skip("DSA signing is not allowed");
```
Comment (slontis) [inline test/acvp_test.c:561]: Does it matter that this returns 123

**Review by slontis — APPROVED** [review]
Almost needs to auto run extended tests whenever there is a fips change?

**Review by t-j-h — APPROVED** [review]

**Comment by t-j-h** [comment]
Agree urgent.

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Agree urgent

File: test/acvp_test.c [hunk 2]
```diff
@@ -357,6 +360,9 @@ static int ecdh_cofactor_derive_test(int tstid)
     int use_cofactordh = t->key_cofactor;
     int cofactor_mode = t->derive_cofactor_mode;
 
+    if (!ec_cofactors)
```
Comment (nhorman) [inline test/acvp_test.c:363]: If OPENSSL_NO_EC is defined, ec_cofactors won't be declared.  Won't the build break here in that case?

**Review by nhorman — CHANGES_REQUESTED** [review]

File: test/acvp_test.c (hunk 2 above)
Comment (paulidale) [reply test/acvp_test.c:363]: Line 139 has the guarding ifdef.

**Comment by paulidale** [comment]
Suggest making the extended provider compatibility test always run, or at least a subset of them: PR vs old FIPS releases?

**Pull request closed without merging**